### PR TITLE
Fi 540 check resource id from read test

### PR DESCRIPTION
**Pull request by czh-orz**
This PR makes it so that validate_read_reply will check if the resource returned has the same id as the resource used for the read test. I don't test the case when the resource is a reference because I didn't know a way to get the id for any reference and I don't think that situation is used in any of our existing sequences.

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-540
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [X] Tests are included and test edge cases
- [X] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/app/sequence_base.rb [hunk 1]
```diff
@@ -524,7 +524,8 @@ def validate_search_reply(klass, reply, search_params)
       def validate_read_reply(resource, klass)
         class_name = klass.name.demodulize
         assert !resource.nil?, "No #{class_name} resources available from search."
-        if resource.is_a? FHIR::DSTU2::Reference
+        resource_is_reference = resource.is_a? FHIR::DSTU2::Reference
```
Comment (arscan) [inline lib/app/sequence_base.rb]: Does this line only work for DSTU2 servers?  The name of the method doesn't seem to indicate it is only for DSTU2.

Comment (arscan) [reply lib/app/sequence_base.rb]: Could you use `resource.is_a? versioned_resource_class('Reference')`

File: lib/app/sequence_base.rb [hunk 2]
```diff
@@ -524,7 +524,8 @@ def validate_search_reply(klass, reply, search_params)
...
+        if resource_is_reference
           read_response = resource.read
         else
           id = resource.try(:id)
```
Comment (arscan) [inline lib/app/sequence_base.rb]: Might as well change this `resource&.id` maybe

**Comment by Jammjammjamm** [comment]
I'm not saying that you need to support getting ids from references, but if you want to, the id is whatever comes after the last `/` in the reference. If we don't ever use references here, though, it would be nice to get rid of that code altogether.

File: lib/app/sequence_base.rb (hunk 2 above)
Comment (czh-orz) [reply lib/app/sequence_base.rb]: Changed.

File: lib/app/sequence_base.rb (hunk 1 above)
Comment (czh-orz) [reply lib/app/sequence_base.rb]: Removed logic for reference altogether since it's not being used anywhere.

**Comment by czh-orz** [comment]
> I'm not saying that you need to support getting ids from references, but if you want to, the id is whatever comes after the last `/` in the reference. If we don't ever use references here, though, it would be nice to get rid of that code altogether.

I just removed the reference part.

**Comment by czh-orz** [comment]
I added the reference check back in because it's being used in one test in the Argonaut MedicationOrder sequence.

**Pull request merged**